File: Add Code Here/PYTHON/1-two-sum/1-two-sum.py
#Time: O(n)
#Space: O(n)
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        # create a dictionary to store the values
        d = {}
        
        # for each value in the array
        for i, value in enumerate(nums):
            # if the value is in the dictionary
            if target - value in d:
                return d[target - value], i
            else:
                d[value] = i
        return None

File: Hacktoberfest2023-main/Add Code Here/LeetCode Problems/1. Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        for(int i=0;i<nums.size();i++)
        {
            for(int j=i+1;j<nums.size();j++)
            {
                if(nums[i]+nums[j]==target)
                return {i,j};
            }
        }
        return {};
    }
};
